### ISSUE TITLE
CTRL+D problem replacing consecutive characters

### ISSUE THREAD

**Issue by brunoscopelliti** [Microsoft/vscode#37967]
- VSCode Version: 1.18.0
- OS Version: Windows 10

Reproduces without extensions: Yes

I made a screen recording that reproduce the issue:

![issue](https://user-images.githubusercontent.com/3489757/32614837-67dbd9e8-c56e-11e7-8f71-47292f3cc05f.gif)

I want to replace single quotes with double quotes. 

So I follow these steps:
1.  Select all the occurrences of single quotes using `CTRL+D`.
2. Delete single quotes using backspace.
3. Type the double quotes char.

I expected that also `data-root=''` would have been replaced with `data-root=""`.
It has been replaced with `data-root="` instead.

The same problem is reproducible with whatever couple of consecutive equal characters; 
eg. When I try to replace all the occurences of `x` with `y` in `xfoox xx xbarx` (following the previous steps) I get `yfooy y ybary`.

**Comment by shobhitchittora** [comment]
@brunoscopelliti I think this is the intended behavior. 
Tip: If u want to replace single quotes with double just select all instances of them and press double quotes ( without first deleting them ). 😇

**Comment by brunoscopelliti** [comment]
Hey @shobhitchittora ,
I can agree with you that this is not the worst problem ever, but I still think it is a disturbing behaviour, that in case could be easily addressed, I would be happy to see fixed.
As for the tip that you suggested... 

```js
const a = 'foo'
const b = ''
```
that's what happens to the above snippet, when I select all instances of single quotes and press double quotes (without first deleting them):

```js
const a = "'"foo"'"
const b = "'""'"
```

**Comment by shobhitchittora** [comment]
@brunoscopelliti This is my screengrab following what suggestion in my last comment. Please verify.
![screengrab](https://user-images.githubusercontent.com/4711855/32696632-44b25d98-c7a3-11e7-8821-4948f9b27a1e.gif)

**Comment by brunoscopelliti** [comment]
@shobhitchittora I've the same result that you've recorded when I play in an *untitled* file; but when I try in a real project file (with *.js* extension) the result is the one I've described in my previous comment. Could you please check?

**Comment by shobhitchittora** [comment]
@brunoscopelliti you are indeed right. From what I observe is that vscode is trying to auto-correct ahead for you. This happens when you select some text and type ```"```, ```'```,```[```or```(```.

Similar To: #20891

**Comment by alexandrudima** [comment]
Merged PR #38175 which adds `editor.multiCursorMergeOverlapping` to control this behaviour

**Comment by isidorn** [comment]
I verified this settings works as expected, however instead of us introducing a new setting you could have simply used
`"editor.autoClosingBrackets": false`
